packages/@react-spectrum/toast/src/Toast.tsx

```diff
@@ -78,13 +78,13 @@ function Toast(props: SpectrumToastProps, ref: DOMRef<HTMLDivElement>) {
   let Icon = ICONS[variant];
   let {isFocusVisible, focusProps} = useFocusRing();
 
-  const handleAction = () => {
+  const handleAction = (evt: PressEvent) => {
     if (onAction) {
       onAction();
     }
 
     if (shouldCloseOnAction) {
-      state.close(key);
+      closeButtonProps.onPress(evt);
```

Comment:
but we aren't pressing the close button, so this is a false event we're firing?

Reply:
I don't want to duplicate the focus management logic from here: https://github.com/adobe/react-spectrum/pull/6011/files#diff-a1ee157fac4bb14b926cd23ba0263284fa555fbbc8168ed4514858ddda8b583bR103-R115

Reply:
Can we handle the check/move focus on Toast unmount instead? Other design systems may decide to have two or more actions or will maybe close a toast for some other reason while focus is within it.